src/header/parsing.rs

```diff
@@ -23,24 +23,18 @@ pub fn from_raw_str<T: str::FromStr>(raw: &[u8]) -> ::Result<T> {
 
 /// Reads a comma-delimited raw header into a Vec.
 #[inline]
-pub fn from_comma_delimited<T: str::FromStr>(raw: &[Vec<u8>]) -> ::Result<Vec<T>> {
-    if raw.len() != 1 {
-        return Err(::Error::Header);
+pub fn from_comma_delimited<T: str::FromStr, S: AsRef<[u8]>>(raw: &[S]) -> ::Result<Vec<T>> {
+    let mut result = Vec::new();
```

Comment:
I'd like to believe that `httparse` will error if there is a header name with no value, such as `Host:\r\nContent-Length: 10\r\n\r\n`. Which, if that's the case, then `raw` could never be 0.